util/avu.py

```diff
@@ -21,18 +21,70 @@
 Avu.unit  = Avu.u
 
 
+def of_data(ctx, path):
+    """Get (a,v,u) triplets for a given data object."""
+    return itertools.imap(lambda x: Avu(*x),
+                          genquery.Query(ctx, "META_DATA_ATTR_NAME, META_DATA_ATTR_VALUE, META_DATA_ATTR_UNITS",
+                                              "COLL_NAME = '{}' AND DATA_NAME = '{}'".format(*pathutil.chop(path))))
+
+
 def of_coll(ctx, coll):
     """Get (a,v,u) triplets for a given collection."""
     return itertools.imap(lambda x: Avu(*x),
                           genquery.Query(ctx, "META_COLL_ATTR_NAME, META_COLL_ATTR_VALUE, META_COLL_ATTR_UNITS",
                                               "COLL_NAME = '{}'".format(coll)))
 
 
-def of_data(ctx, path):
-    """Get (a,v,u) triplets for a given data object."""
-    return itertools.imap(lambda x: Avu(*x),
-                          genquery.Query(ctx, "META_DATA_ATTR_NAME, META_DATA_ATTR_VALUE, META_DATA_ATTR_UNITS",
-                                              "COLL_NAME = '{}' AND DATA_NAME = '{}'".format(*pathutil.chop(path))))
+def inside_coll(ctx, path, recursive=False):
+    """Get a list of all AVUs inside a collection with corresponding paths.
+
+    Note: the returned value is a generator / lazy list, so that large
+          collections can be handled without keeping everything in memory.
+          use list(...) on the result to get an actual list if necessary.
+
+    The returned paths are absolute paths (e.g. '/tempZone/home/x').
+
+    :param ctx:       Combined type of a callback and rei struct
+    :param path:      Path of collection
+    :param recursive: List AVUs recursively
+
+    :returns: List of all AVUs inside a collection with corresponding paths
+    """
+    # coll+name -> path
+    def to_absolute(row, type):
+        if type == "collection":
+            return (row[1], type, row[2], row[3], row[4])
+        else:
+            return ('{}/{}'.format(row[0], row[1]), type, row[2], row[3], row[4])
+
+    collection_root = genquery.row_iterator(
+        "COLL_PARENT_NAME, COLL_NAME, META_COLL_ATTR_NAME, META_COLL_ATTR_VALUE, META_COLL_ATTR_UNITS",
```

Comment:
I'm not familiar with what changes the speeds of genquery operations, or all the genquery quirks, but do we need to give back COLL_PARENT_NAME here?

Reply:
I don't think it has much impact on the speed because it is needed in the where condition.

Did it this way to be able to reuse to_absolute for all queries.